src/lib/transition_frontier/snark_pool_refcount.ml

```diff
@@ -12,18 +13,57 @@ module type Inputs_intf = sig
      and type staged_ledger := Staged_ledger.t
 end
 
-module type S = Transition_frontier_extension_intf
-
 module Make (Inputs : Inputs_intf) :
-  S
+  Transition_frontier_extension_intf
   with type transition_frontier_breadcrumb := Inputs.Breadcrumb.t
-   and type input := unit = struct
+   and type input := unit
+   and type view := int Inputs.Transaction_snark_work.Statement.Table.t =
+struct
   module Work = Inputs.Transaction_snark_work.Statement
 
-  type t = {ref_table: int Work.Table.t}
+  type t = int Work.Table.t
+
+  let get_work (breadcrumb : Inputs.Breadcrumb.t) : Work.t Sequence.t =
+    let ledger = Inputs.Breadcrumb.staged_ledger breadcrumb in
+    let scan_state = Inputs.Staged_ledger.scan_state ledger in
+    let work_to_do =
+      Inputs.Staged_ledger.Scan_state.all_work_to_do scan_state
```

Comment:
Is this all the SNARKs left to be generated, or does it include SNARKs that are already in the scan state somewhere?

Reply:
cc @deepthiskumar

Reply:
This is all the computations in the current state for which snarks need to be generated. You can think of scan state as a tree of "jobs". When a snark(or a transaction) is added to the tree, a new "job" is created for which a snark has to be generated.